Add docs for `{NavigateEvent,PopStateEvent}.hasUAVisualTransition` (#36705)
mdn/content: files/en-us/web/api/popstateevent/hasuavisualtransition/index.md

---
title: "PopStateEvent: hasUAVisualTransition property"
short-title: hasUAVisualTransition
slug: Web/API/PopStateEvent/hasUAVisualTransition
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.PopStateEvent.hasUAVisualTransition
---

{{APIRef("History API")}}{{SeeCompatTable}}

> The **`hasUAVisualTransition`** read-only property of the {{domxref("PopStateEvent")}} interface returns `true` if the navigation has been performed a visual transition before dispatching this event by the user agent, or `false` otherwise.

Author: ```suggestion
The **`hasUAVisualTransition`** read-only property of the {{domxref("PopStateEvent")}} interface returns `true` if the user agent performed a visual transition for this navigation before dispatching this event, or `false` otherwise.
```

> Some user agents may provide its builtin visual transition when executing site navigations. If the site author also add a visual transition, these transitions may conflict and cause confusion for the user. The property is intended to avoid the site author and the user agent both add visual transitions, and ensure only one visual transition is executed at a time to make sure good user experience.

Author: ```suggestion
User agents may provide a built-in visual transition when executing site navigations. If the site author also adds a visual transition, user agent and author transitions may conflict and confuse a visitor. The property lets you detect whether a UA transition was provided so that you can skip author transitions for a better user experience.
```

> Some user agents may provide its builtin visual transition when executing site navigations. If the site author also add a visual transition, these transitions may conflict and cause confusion for the user. The property is intended to avoid the site author and the user agent both add visual transitions, and ensure only one visual transition is executed at a time to make sure good user experience.

Reviewer: Same comment as above

## Value

A boolean value.

## Examples

```js
window.addEventListener("popstate", (event) => {
  // Fetch the new content

>   const newContent = await fetchNewContent(event.destination.url, {
>     signal: event.signal,
>   });

Author: ```suggestion
  const newContent = await fetchNewContent(location.href);
```

>   const newContent = await fetchNewContent(event.destination.url, {

Reviewer: Is the origin important here given the event type? The same-origin clarification was good above.

Author: argee...

Reviewer: I mean is this relevant for `popstate`:

```
if (!event.canIntercept) {
    return;
  }
```

Author: yes, I've fixed the examplecode, as the event type is `PopStateEvent`, which only supports `state` and `hasUAVisualTransition`  properties
